### PR TITLE
Promote the `consents` module from `experimental` to `globus_sdk.scopes.consents`

### PR DESCRIPTION
**Pull request by sirosen**
_Based off of #1046_

This promotes `consents` out of experimental and makes a minimal doc addition.
Aliasing from `experimental` to the new code location is included.

----
📚 Documentation preview 📚: https://globus-sdk-python--1047.org.readthedocs.build/en/1047/

### CODE REVIEW

File: docs/authorization/scopes_and_consents/consents.rst [hunk 1]
```diff
@@ -12,6 +12,23 @@ Consents are modeled as a ``ConsentForest`` full of ``ConsentTrees`` containing
 provided by a user to client applications for token grants under certain scoped
 contexts.
 
+Consent objects are provided from ``globus_sdk.scopes.consents``.
+
+They are typically produced by calling
+:meth:`globus_sdk.AuthClient.get_consents` and invoking ``to_forest()`` on
+the response. Example usage:
```
Comment (derek-globus) [inline docs/authorization/scopes_and_consents/consents.rst:19]:
```suggestion
While the consent model classes themselves are exposed here in ``globus_sdk.scopes.consents``,
most objects are actually loaded from a :meth:`globus_sdk.AuthClient.get_consents` response, using the attached :meth:`globus_sdk.GetConsentsResponse.to_forest()` method:
```

**Review by derek-globus — CHANGES_REQUESTED** [review]
One phrasing suggestion but otherwise looks good.

File: docs/authorization/scopes_and_consents/consents.rst (hunk 1 above)
Comment (sirosen) [reply docs/authorization/scopes_and_consents/consents.rst:19]: Good rephrase! I think it has to be written without the parens,
```
:meth:`globus_sdk.GetConsentsResponse.to_forest`
```
but I'm going to commit the suggestion to see if it renders. I'll amend if it fails.

Comment (sirosen) [reply docs/authorization/scopes_and_consents/consents.rst]: I stand corrected -- both formats work fine!

**Review by derek-globus — APPROVED** [review]

**Pull request merged**